embedded-graphics/src/primitives/mod.rs

```diff
@@ -41,6 +44,111 @@ pub trait ContainsPoint {
     fn contains(&self, point: Point) -> bool;
 }
 
+/// Create an [`Arc`](./primitives/arc/struct.Arc.html) with optional styling using a
+/// convenient macro.
+///
+/// ```rust
+/// use embedded_graphics::{
+///     egarc,
+///     pixelcolor::Rgb565,
+///     prelude::*,
+///     primitive_style,
+///     primitives::Arc,
+///     style::{PrimitiveStyle, Styled},
+/// };
+///
+/// // Coordinates can be defined as any type that implements `Into<Point>`
+/// // Angles can be defined as any type that implements `Into<Angle>`
+/// let line_arc: Styled<Arc, PrimitiveStyle<Rgb565>> =
+///     egarc!(top_left = (10, 20), diameter = 30, angle_start = 90.0, angle_sweep = 60.0);
+///
+/// let filled_arc: Styled<Arc, PrimitiveStyle<Rgb565>> = egarc!(
+///     center = (10, 20),
+///     diameter = 30,
+///     angle_start = 90.0,
+///     angle_sweep = 60.0,
+///     style = primitive_style!(stroke_color = Rgb565::RED, fill_color = Rgb565::GREEN)
+/// );
+/// ```
+///
+/// Style properties like `stroke_color` map to methods on the [`PrimitiveStyleBuilder`] struct.
+/// For example, the following code makes two identical arcs:
+///
+/// [`PrimitiveStyleBuilder`]: style/struct.PrimitiveStyleBuilder.html
+///
+/// ```rust
+/// use embedded_graphics::{
+///     egarc,
+///     pixelcolor::Rgb565,
+///     prelude::*,
+///     primitive_style,
+///     primitives::Arc,
+///     style::{PrimitiveStyle, PrimitiveStyleBuilder, Styled},
+/// };
+///
+/// let arc_1: Styled<Arc, PrimitiveStyle<Rgb565>> = egarc!(
+///     top_left = (10, 20),
+///     diameter = 30,
+///     angle_start = 90.0,
+///     angle_sweep = 60.0,
+///     style = primitive_style!(
+///         stroke_color = Rgb565::RED,
+///         fill_color = Rgb565::GREEN,
+///         stroke_width = 1
+///     )
+/// );
+///
+/// let style = PrimitiveStyleBuilder::new()
+///     .fill_color(Rgb565::GREEN)
+///     .stroke_color(Rgb565::RED)
+///     .stroke_width(1)
+///     .build();
+///
+/// let arc_2: Styled<Arc, PrimitiveStyle<Rgb565>> =
+///     Arc::new(Point::new(10, 20), 30, Angle::from_degree(90.0), Angle::from_degree(60.0)).into_styled(style);
+///
+/// assert_eq!(arc_1, arc_2);
+/// ```
+#[macro_export]
+macro_rules! egarc {
```

Comment:
IMO the macro shouldn't convert from degree to `Angle` internally, but take `Angle` as an argument instead. This way the user can use `180.0.deg()`, `PI.rad()` and even `some_angle_variable + 123.0.deg()` as angles.

Reply:
Yes, you're right. I didn't put much thought about those macros after adding support for floating point angles and radiant. Changed to take `Angle` argument now.